### PR TITLE
Implemented Fast Layered Copy

### PR DESCRIPTION
**Pull request by FernandoS27**
This fixes performance regressions in ARMS and Splatoon and improves loading performance in Many Games.

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp [hunk 1]
```diff
@@ -1275,6 +1275,31 @@ Surface RasterizerCacheOpenGL::GetUncachedSurface(const SurfaceParams& params) {
     return surface;
 }
 
+void RasterizerCacheOpenGL::FastLayeredCopySurface(const Surface& src_surface,
+                                                   const Surface& dst_surface) {
+    const auto& init_params{src_surface->GetSurfaceParams()};
+    const auto& dst_params{dst_surface->GetSurfaceParams()};
+    VAddr address = init_params.addr;
+    std::size_t layer_size = dst_params.LayerMemorySize();
```
Comment (lioncash) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: This can be const

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp [hunk 2]
```diff
@@ -1275,6 +1275,31 @@ Surface RasterizerCacheOpenGL::GetUncachedSurface(const SurfaceParams& params) {
...
+    for (u32 layer = 0; layer < dst_params.depth; layer++) {
+        for (u32 mipmap = 0; mipmap < dst_params.max_mip_level; mipmap++) {
+            VAddr s_address = address + dst_params.GetMipmapLevelOffset(mipmap);
```
Comment (lioncash) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: Ditto

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp [hunk 3]
```diff
@@ -1275,6 +1275,31 @@ Surface RasterizerCacheOpenGL::GetUncachedSurface(const SurfaceParams& params) {
...
+            auto copy = TryGet(s_address);
```
Comment (lioncash) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: Ditto

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp (hunk 2 above)
Comment (Subv) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: what is 's_address'? use a more descriptive name

**Pull request merged**